### PR TITLE
Fixes #7240: The rudder_var path on Solaris is incorrect

### PR DESCRIPTION
**Pull request by Kegeruneku**
https://www.rudder-project.org/redmine/issues/7240

### CODE REVIEW

File: techniques/system/common/1.0/site.st [hunk 1]
```diff
@@ -57,9 +58,6 @@ bundle common g
       "crontab"                    string => "/etc/crontab";
       "gzip"                       string => "/bin/gzip";
 
-      # DEPRECATED: This variable is used in pre-2.9 Techniques.
```
Comment (Kegeruneku) [inline techniques/system/common/1.0/site.st]: Now's the good time to get rid of this crap

File: techniques/system/inventory/1.0/fusionAgent.st [hunk 2]
```diff
@@ -142,7 +142,7 @@ bundle common inventory
     android::
       "cpuid_tool"      string => "${g.rudder_tools}/cpuid-android-V1.0.sh";
 
-    !aix.!windows.!android::
+    !aix.!solaris.!windows.!android::
```
Comment (Kegeruneku) [inline techniques/system/inventory/1.0/fusionAgent.st]: Won't work on Solaris

File: techniques/system/common/1.0/promises.st [hunk 3]
```diff
@@ -97,20 +97,24 @@ bundle common va
...
 
-    !android.!windows::
-      "rudder_var"     string => "/var/rudder";
+    windows::
+      "rudder_base"    string => "${sys.winprogdir}\Rudder";
+      "rudder_tools"   string => "${rudder_base}\sbin";
+      "ncf_path"       string => "${rudder_base}\ncf";
+
+    solaris::
+      "rudder_var"     string => "/opt/rudder/var";
```
Comment (peckpeck) [inline techniques/system/common/1.0/promises.st]: this should be /var/rudder

File: techniques/system/common/1.0/site.st [hunk 4]
```diff
@@ -33,15 +33,16 @@ bundle common g
       "rudder_rm"                  string => "/system/xbin/rm";
       "rudder_tools_files_android" slist => { "cpuid-android-V1.0.sh" };
 
-    !windows.!android::
+    solaris::
       "rudder_base"                string => "/opt/rudder";
+      "rudder_var"                 string => "/opt/rudder/var";
```
Comment (peckpeck) [inline techniques/system/common/1.0/site.st]: Here too

**Comment by ncharles** [comment]
This PR needs to change the path in the failsafe.st as well

**Comment by ncharles** [comment]
Should target branch 3.2

**Comment by ncharles** [comment]
replaced by https://github.com/Normation/rudder-techniques/pull/788/files

**Pull request closed without merging**